src/navigation/decorate.js

```diff
@@ -0,0 +1,61 @@
+"use strict";
+
+function isCurrentLink(item, currentUrl){
+	return (item.href && item.href === currentUrl) || (item.id && currentUrl.includes(item.id));
+}
+
+function decorateDrawerItem(item, currentUrl){
```

Comment:
I don't these methods are specific to the drawer - they should be able to traverse any link structure